modules/images/dominant-color/load.php

```diff
@@ -88,6 +88,12 @@ function dominant_color_img_tag_add_dominant_color( $filtered_image, $context, $
 		return $filtered_image;
 	}
 
+	// Only apply the dom color to images that have a src attribute that
+	// starts with a double quote, ensuring escaped JSON is also excluded.
+	if ( false === strpos( $filtered_image, ' src="' ) ) {
```

Comment:
Would be great to add a unit test for this:
* One `img` without any `src` attribute (should not be modified).
* One `img` which has its attributes (including `src`) using single quotes instead of double quotes.